python/kserve/test/test_infer_type.py

```diff
@@ -0,0 +1,344 @@
+                                                       "test-int": InferParameter(int64_param=100)
+                                                   },
+                                               }]
+                                           )
+            expected = InferResponse(model_name="TestModel", response_id="123",
+                                     parameters={
+                                         "test-str": InferParameter(string_param="dummy"),
+                                         "test-bool": InferParameter(bool_param=True),
+                                         "test-int": InferParameter(int64_param=100)
+                                     },
+                                     infer_outputs=[
+                                         InferOutput(name="output-0", datatype="INT32", shape=[1, 2], data=[1, 2],
+                                                     parameters={
+                                                         "test-str": InferParameter(string_param="dummy"),
+                                                         "test-bool": InferParameter(bool_param=True),
+                                                         "test-int": InferParameter(int64_param=100)
+                                                     })],
+                                     from_grpc=True
+                                     )
+            res = InferResponse.from_grpc(infer_res)
+            assert InferResponseMatcher(expected).__eq__(res) is True
+
+
+class InferInputMatcher:
+    def __init__(self, infer_input: InferInput):
+        self.input = infer_input
+
+    def __eq__(self, other):
+        if not isinstance(other, InferInput):
+            return False
+        if self.input.name != other.name:
+            return False
+        if self.input.shape != other.shape:
+            return False
+        if self.input.datatype != other.datatype:
+            return False
+        if self.input.parameters != other.parameters:
+            return False
+        if self.input.data != other.data:
+            return False
+        return True
+
+
+class InferRequestMatcher:
+    def __init__(self, req: InferRequest):
+        self.req = req
+
+    def __eq__(self, other):
+        if not isinstance(other, InferRequest):
+            return False
+        if self.req.model_name != other.model_name:
+            return False
+        if self.req.id != other.id:
+            return False
+        if self.req.from_grpc != other.from_grpc:
+            return False
+        if self.req.parameters != other.parameters:
+            return False
+        if len(self.req.inputs) != len(other.inputs):
+            return False
+        for i in range(len(self.req.inputs)):
+            if not InferInputMatcher(self.req.inputs[i]).__eq__(other.inputs[i]):
+                return False
+        return True
+
+
+class InferOutputMatcher:
```

Comment:
Same here should be defined in InferOutput

Reply:
done